src/lib/support/jsontlv/TextFormat.h

```diff
@@ -0,0 +1,51 @@
+/*
+ *
+ *    Copyright (c) 2020 Project CHIP Authors
+ *    Copyright (c) 2013-2017 Nest Labs, Inc.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <algorithm>
+#include <json/json.h>
+#include <lib/core/TLV.h>
+#include <string>
+
+namespace chip {
+/*
+ * Pretty-prints the input Json string using standard library pretty-printer.
+ * This pretty-printer generates a Json string in a human friendly format with 3 space indentation
+ * and nice representation of arrays and objects.
+ * The input can be any string, as long as it's valid Json.
+ */
+std::string PrettyPrintJsonString(const std::string & jsonString)
+{
+    Json::Reader reader;
```

Comment:
Why is this implementation inlined?

Reply:
it should not be inline, should be separate. updated, thanks